src/main/java/io/strimzi/kafka/bridge/http/HttpAdminClientEndpoint.java

```diff
@@ -157,4 +166,118 @@ public void doGetTopic(RoutingContext routingContext) {
+                        describeTopicsResult.cause().getMessage()
+                );
+                HttpUtils.sendResponse(routingContext, HttpResponseStatus.NOT_FOUND.code(),
+                        BridgeContentType.KAFKA_JSON, error.toJson().toBuffer());
+            } else {
+                HttpBridgeError error = new HttpBridgeError(
+                        HttpResponseStatus.INTERNAL_SERVER_ERROR.code(),
+                        describeTopicsResult.cause().getMessage()
+                );
+                HttpUtils.sendResponse(routingContext, HttpResponseStatus.INTERNAL_SERVER_ERROR.code(),
+                        BridgeContentType.KAFKA_JSON, error.toJson().toBuffer());
+            }
+        });
+    }
+
+    public void doGetPartition(RoutingContext routingContext) {
+        String topicName = routingContext.pathParam("topicname");
+        final int partitionId;
+        try {
+            partitionId = Integer.parseInt(routingContext.pathParam("partitionid"));
+        } catch (NumberFormatException ne) {
+            HttpBridgeError error = new HttpBridgeError(
+                    HttpResponseStatus.UNPROCESSABLE_ENTITY.code(),
+                    "Specified partition is not a valid number");
+            HttpUtils.sendResponse(routingContext, HttpResponseStatus.UNPROCESSABLE_ENTITY.code(),
+                    BridgeContentType.KAFKA_JSON, error.toJson().toBuffer());
+            return;
+        }
+        describeTopics(Collections.singletonList(topicName), describeTopicsResult -> {
+            if (describeTopicsResult.succeeded()) {
+                Map<String, TopicDescription> topicDescriptions = describeTopicsResult.result();
+                JsonObject root = null;
+                TopicDescription description = topicDescriptions.get(topicName);
+                if (description != null && partitionId < description.getPartitions().size()) {
+                    TopicPartitionInfo partitionInfo = description.getPartitions().get(partitionId);
+                    int leaderId = partitionInfo.getLeader().getId();
+                    root = new JsonObject();
+                    root.put("partition", partitionInfo.getPartition());
+                    root.put("leader", leaderId);
+                    JsonArray replicasArray = new JsonArray();
+                    Set<Integer> insyncSet = new HashSet<Integer>();
+                    partitionInfo.getIsr().forEach(node -> {
+                        insyncSet.add(node.getId());
+                    });
+                    partitionInfo.getReplicas().forEach(node -> {
+                        JsonObject replica = new JsonObject();
+                        replica.put("broker", node.getId());
+                        replica.put("leader", leaderId == node.getId());
+                        replica.put("in_sync", insyncSet.contains(node.getId()));
+                        replicasArray.add(replica);
+                    });
```

Comment:
the previous 10 lines (or so) of code are the same as the ones in the `doListPartitions` for filling the partition metadata JSON object. Can we factor out a common method to use in both?

Reply:
ok